### PR TITLE
Implement detection mAP evaluator wrapper and unify label format between SSD loss and mAP evaluator 

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/8735

### CODE REVIEW

File: python/paddle/fluid/evaluator.py [hunk 1]
```diff
@@ -285,3 +287,108 @@ def eval(self, executor, eval_program=None):
...
+
+    Example:
+
+        exe = fluid.executor(place)
+        map_evaluator = fluid.Evaluator.DetectionMAP(input, label)
+        cur_map, accum_map = map_evaluator.get_map_var()
+        for epoch in PASS_NUM:
+            map_evaluator.reset(exe)
+            for data in batches:
+                loss, cur_map_v, accum_map_v = exe.run(fetch_list=[cost, cur_map, accum_map])
```
Comment (pkuyym) [inline python/paddle/fluid/evaluator.py]: Seems exceeded 80 columns.

Comment (qingqing01) [reply python/paddle/fluid/evaluator.py]: Done.

File: paddle/fluid/operators/detection_map_op.cc [hunk 2]
```diff
@@ -96,6 +95,10 @@ class DetectionMAPOpMaker : public framework::OpProtoAndCheckerMaker {
              "instance, the offsets in first dimension are called LoD, "
              "the number of offset is N + 1, if LoD[i + 1] - LoD[i] == 0, "
              "means there is no ground-truth data.");
+    AddInput("HasState",
+             "(Tensor<int>) A tensor with shape [1], 0 mean ignoring input "
```
Comment (pkuyym) [inline paddle/fluid/operators/detection_map_op.cc]: mean --> means

File: python/paddle/fluid/evaluator.py [hunk 3]
```diff
@@ -285,3 +287,120 @@ def eval(self, executor, eval_program=None):
...
+
+class DetectionMAP(Evaluator):
+    """
+    Calculate the detection mean average precision (mAP).
+
+    TODO (Dang Qingqing): update the following doc.
+    The general steps are as follows:
+    1. calculate the true positive and false positive according to the input
+        of detection and labels.
+    2. calculate mAP value, support two methods: '11 point' and 'integral'.
```
Comment (pkuyym) [inline python/paddle/fluid/evaluator.py]: methods --> versions

File: python/paddle/fluid/evaluator.py [hunk 4]
```diff
@@ -285,3 +287,120 @@ def eval(self, executor, eval_program=None):
...
+
+    Please get more information from the following articles:
+      https://sanchom.wordpress.com/tag/average-precision/
+      https://arxiv.org/abs/1512.02325
+
+    Args:
+        input (Variable): the detection results, which is a LoDTensor with shape
```
Comment (pkuyym) [inline python/paddle/fluid/evaluator.py]: the --> The ? same as follows.

File: paddle/fluid/operators/detection_map_op.cc (hunk 2 above)
Comment (qingqing01) [reply paddle/fluid/operators/detection_map_op.cc]: Done. Thanks!

File: python/paddle/fluid/evaluator.py (hunk 3 above)
Comment (qingqing01) [reply python/paddle/fluid/evaluator.py]: Done. Thanks!

File: python/paddle/fluid/evaluator.py (hunk 4 above)
Comment (qingqing01) [reply python/paddle/fluid/evaluator.py]: Done. From http://www.paddlepaddle.org/docs/develop/api/en/fluid/layers.html , upper case is betther.

**Pull request merged**